mlflow/server/js/src/experiment-tracking/pages/prompts/components/ModelConfigForm.tsx

```diff
@@ -1,19 +1,32 @@
 import {
+  DialogCombobox,
+  DialogComboboxContent,
+  DialogComboboxOptionList,
+  DialogComboboxOptionListSearch,
+  DialogComboboxOptionListSelectItem,
+  DialogComboboxTrigger,
   FormUI,
   InfoSmallIcon,
   Popover,
   RHFControlledComponents,
   Typography,
   useDesignSystemTheme,
 } from '@databricks/design-system';
-import { useFormContext } from 'react-hook-form';
+import { useMemo, useEffect, useRef } from 'react';
+import { useFormContext, Controller } from 'react-hook-form';
 import { FormattedMessage, useIntl } from 'react-intl';
+import { useProvidersQuery } from '../../../../gateway/hooks/useProvidersQuery';
+import { useModelsQuery } from '../../../../gateway/hooks/useModelsQuery';
+
+const PRIORITY_PROVIDERS = ['openai', 'anthropic', 'gemini', 'databricks'];
```

Comment:
It feels a bit odd to purely sort by alphabetic order, so I am pulling these 4 providers to the front.